web/src/util.ts

```diff
@@ -629,3 +629,12 @@ export async function sha256_hash(text: string): Promise<string | undefined> {
     const hashHex = hashArray.map((byte) => byte.toString(16).padStart(2, "0")).join("");
     return hashHex;
 }
+
+export function unique_array_insert<T>(array: T[], new_item: T): void {
+    for (const item of array) {
+        if (_.isEqual(item, new_item)) {
+            return;
+        }
+    }
+    array.push(new_item);
+}
```

Comment:
This should have a comment noting it should only be used in loops with small collections, since it runs in linear time. 

(Better would be to use a data structure without this risk).